src/main/java/org/minions/devfund/bruno/battleship/Ocean.java

```diff
@@ -0,0 +1,234 @@
+package org.minions.devfund.bruno.battleship;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+import static java.util.stream.IntStream.range;
+
+/**
+ * Class to in charge that manage the ocean event.
+ */
+public class Ocean {
+    private static final int SIZE = 20;
+    private static final int TOTAL_SHIPS = 13;
+    private static final String BATTLESHIP = "battleship";
+    private static final String BATTLECRUISER = "battlecruiser";
+    private static final String CRUISER = "cruiser";
+    private static final String LIGHTCRUISER = "lightcruiser";
+    private static final String DESTROYER = "destroyer";
+    private static final String SUBMARINE = "submarine";
+    private Ship[][] ships = new Ship[SIZE][SIZE];
+    private int shotsFired;
+    private int hitCount;
+    private int shipsSunk;
+
+    /**
+     * Constructor that initializes any game variables and generates the empty ocean.
+     */
+    public Ocean() {
+        shotsFired = 0;
+        hitCount = 0;
+        shipsSunk = 0;
+        generateOcean();
+    }
+
+    /**
+     * Generates a empty ocean.
+     */
+    private void generateOcean() {
+        range(0, SIZE).forEach(row -> range(0, SIZE).forEach(col -> {
+            EmptySea emptySea = new EmptySea();
+            emptySea.placeShipAt(row, col, true, this);
+        }));
+    }
+
+    /**
+     * Place all randomly on the (initially empty) ocean.
+     * Place larger ships before smaller ones, or you may end up with no legal place to put a large ship.
+     */
+    public void placeAllShipsRandomly() {
+        List<String> shipNames = Arrays.asList(BATTLESHIP, BATTLECRUISER, CRUISER, CRUISER, LIGHTCRUISER,
+                LIGHTCRUISER, DESTROYER, DESTROYER, DESTROYER, SUBMARINE, SUBMARINE, SUBMARINE, SUBMARINE);
+        List<Ship> ships = new LinkedList<>();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "ships" which hides the field declared at line 24. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)